plugins/cinema4d/import_cast.pyp

```diff
@@ -0,0 +1,431 @@
+import c4d
+import os
+import array
+import math
+from c4d import plugins, bitmaps, Vector, gui, BaseObject
+import mxutils
+
+resDir = os.path.join(os.path.dirname(__file__), "res")
+mxutils.ImportSymbols(resDir)
+with mxutils.LocalImportPath(resDir):
+    from cast import Cast, CastColor, Model, Animation, Instance, File
+
+__pluginname__ = "Cast"
+
+class CastLoader(plugins.SceneLoaderData):
+    def Identify(self, node, name, probe, size):
+        if "cast" in name[-4:]:
+            return True
+        return False
+
+    def Load(self, node, name, doc, filterflags, error, bt):
+        importCast(doc, node, name)
+        return c4d.FILEERROR_NONE
+
+def importCast(doc, node, path):
+    cast = Cast.load(path)
+
+    instances = []
+
+    for root in cast.Roots():
+        for child in root.ChildrenOfType(Model):
+            importModelNode(doc, node, child, path)
+        for child in root.ChildrenOfType(Animation):
+            importAnimationNode(doc, child, path)
+        for child in root.ChildrenOfType(Instance):
+            instances.append(child)
+
+    if len(instances) > 0:
+        importInstanceNodes(doc, instances, path)
+        
+"""
+We are using Blender unpack_list for convinience
+https://github.com/blender/blender/blob/main/scripts/modules/bpy_extras/io_utils.py#L370
+"""
+def unpack_list(list_of_tuples):
+    flat_list = []
+    flat_list_extend = flat_list.extend  # a tiny bit faster
+    for t in list_of_tuples:
+        flat_list_extend(t)
+    return flat_list
+
+def utilityQuaternionToEuler(q):
+    x, y, z, w = q
```

Comment:
I've done this, cause I had struggles with c4d.Quaternions. Everything I tried gave me a result of none or that I need a matrix input. I think I know why now:

From an official example that came out 2 days ago:
"The setter c4d.Quaternion.v is currently bugged. This will be fixed in a future version of
      Cinema 4D. But this makes it currently impossible to construct a c4d.Quaternion from the
      maxon::Quaternion32 data exposed by the particle system. As a workaround, we can construct
      a particle alignment from the particle velocity."
https://github.com/PluginCafe/cinema4d_py_sdk_extended/blob/master/scripts/04_3d_concepts/simulation/particle_group_object_2024_4.py#L17-L20

Reply:
Well that's crazy! Good to note.